test/test_channel.lua

```diff
@@ -0,0 +1,402 @@
+local test = require 'regress'
+local ipc = require 'libipc'
+
+test {
+   -- it should be possible to open a channel, write something to it
+   -- and read it back within the same thread.
+   openReadWriteSameThread = function()
+      local c = ipc.channel()
+      local data = 10
+      local status = c:write(data)
+      test.mustBeTrue(status == ":open")
```

Comment:
I was thinking about this, but I'm not sure it would work because of the following reasons.

1. If you can write and read nils into a channel, then you can't return nil from read or write to signify failure to read/write a value.
2. For non-blocking reads/writes you want to know whether the read/write succeeded and why.. and for that you need a status return variable.
3. For blocking reads/writes, you would be interested in knowing whether the channel was closed for that write or whether the channel was drained for that read. Otherwise it's not clear whether your value was written to the channel and it would not be clear whether a nil was because nil was read or because the channel was drained.

Maybe I could replace the string statuses with some kind of enum, so that the status is an enum and then you can compare that to ipc.channel.OPEN. i.e. instead of ```status == ":open"``` it would be  ```status == ipc.channel.OPEN```. I'm not sure if that is what you were suggesting in the first place.